src/main/java/cs2103/v15_1j/jimjim/ui/SearchPaneController.java

```diff
@@ -87,16 +89,15 @@ private void showSearchResults(){
 			}
 		}
 
-		counter = 0;
-		for(DeadlineTask task: lists.getDeadlineTasksList()){
-			counter++;
+		for(DeadlineTask task: searchResultsList.getDeadlineTasksList()){
 			CheckBox cb = new CheckBox();
 			cb.selectedProperty().bindBidirectional(task.completedProperty());
 			cb.setDisable(true);
 			GridPane.setHalignment(cb, HPos.CENTER);
 			searchGridPane.addColumn(0, cb);
 
-			Label idLabel = new Label("[D"+counter+"]");
+			int id = masterList.getDeadlineTasksList().indexOf(task) + 1;
```

Comment:
no need. just use `masterList.indexOf` it will automatically search the right list based on the class